sdks/java/io/iceberg/src/main/java/org/apache/beam/sdk/io/iceberg/IcebergIO.java

```diff
@@ -413,29 +438,56 @@ abstract static class Builder {
 
       abstract Builder setTableIdentifier(TableIdentifier identifier);
 
+      abstract Builder setFromSnapshotExclusive(@Nullable Long fromSnapshotExclusive);
+
+      abstract Builder setToSnapshot(@Nullable Long toSnapshot);
+
+      abstract Builder setTriggeringFrequency(Duration triggeringFrequency);
+
       abstract ReadRows build();
     }
 
     public ReadRows from(TableIdentifier tableIdentifier) {
       return toBuilder().setTableIdentifier(tableIdentifier).build();
     }
 
+    public ReadRows fromSnapshotExclusive(@Nullable Long fromSnapshotExclusive) {
+      return toBuilder().setFromSnapshotExclusive(fromSnapshotExclusive).build();
+    }
+
+    public ReadRows toSnapshot(@Nullable Long toSnapshot) {
+      return toBuilder().setToSnapshot(toSnapshot).build();
+    }
+
+    public ReadRows withTriggeringFrequency(Duration triggeringFrequency) {
+      return toBuilder().setTriggeringFrequency(triggeringFrequency).build();
+    }
+
     @Override
     public PCollection<Row> expand(PBegin input) {
       TableIdentifier tableId =
           checkStateNotNull(getTableIdentifier(), "Must set a table to read from.");
 
       Table table = getCatalogConfig().catalog().loadTable(tableId);
 
-      return input.apply(
-          Read.from(
-              new ScanSource(
-                  IcebergScanConfig.builder()
-                      .setCatalogConfig(getCatalogConfig())
-                      .setScanType(IcebergScanConfig.ScanType.TABLE)
-                      .setTableIdentifier(tableId)
-                      .setSchema(IcebergUtils.icebergSchemaToBeamSchema(table.schema()))
-                      .build())));
+      IcebergScanConfig scanConfig =
+          IcebergScanConfig.builder()
+              .setCatalogConfig(getCatalogConfig())
+              .setScanType(IcebergScanConfig.ScanType.TABLE)
+              .setTableIdentifier(tableId)
+              .setSchema(IcebergUtils.icebergSchemaToBeamSchema(table.schema()))
+              .setFromSnapshotExclusive(getFromSnapshotExclusive())
+              .setToSnapshot(getToSnapshot())
+              .build();
+      if (getTriggeringFrequency() != null
```

Comment:
I'm not too convinced this should be what controls the incremental scan source. I think it might be best if the user very explicitly says they want to read unbounded rows, versus reading the table as a bounded data set.

Reply:
Outlined the design here: https://s.apache.org/beam-iceberg-incremental-source

But specifically for this comment, here's what the decision tree looks like:
- If none of the above options [frequency, to, from] are set, use the existing (old) bounded scan source.
- If any of these options are set, use the new incremental scan source.
  - If `triggering_frequency_seconds` is set, use the unbounded implementation. 
  - Otherwise, use the bounded implementation.

I think an explicit "streaming=true" parameter has its merits but can be unnecessary, since a triggering frequency is needed anyways to determine the poll interval.

Reply:
OK. I don't love it but this will be a pretty minor part of how a user interacts with it anyhow. Noting that the Flink source has explicit `streaming(true)` 😛 

If we support for incremental reads including deletes for CDC we'll need it to be more explicit and have a different output type.

Reply:
After offline discussions, we decided to make this a new Managed connector (`Managed.ICEBERG_CDC`) that is streaming only. I'll keep the `.streaming(true)` option for IcebergIO itself, but it won't be raised up as a Managed configuration option